core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/loader/ClasspathSdkHttpServiceProvider.java

```diff
@@ -46,22 +48,33 @@
 
     @Override
     public Optional<T> loadService() {
-        Iterator<T> httpServices = serviceLoader.loadServices(serviceClass);
-        if (!httpServices.hasNext()) {
+        Iterable<T> iterable = () -> serviceLoader.loadServices(serviceClass);
+        List<T> impls = StreamSupport
+            .stream(iterable.spliterator(), false)
+            .collect(Collectors.toList());
+
+        if (impls.isEmpty()) {
             return Optional.empty();
         }
-        T httpService = httpServices.next();
 
-        if (httpServices.hasNext()) {
+        if (impls.size() > 1) {
+
+            String implText =
+                impls.stream()
+                     .map(clazz -> clazz.getClass().getName())
+                     .collect(Collectors.joining(",","[","]"));
```

Comment:
Looks like checkstyle wants a space between the arguments here. Can you add those?

Reply:
Whoops, done!